platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+                if (claimType.equalsIgnoreCase(attribute.getName())) {
+                    String claimValue = attribute.getDOM().getTextContent();
+                    if (attributeMap.containsKey(attribute.getDOM().getTextContent())) {
+                        claimsCollection.add(createSingleValuedClaim(claimType,
+                                attributeMap.get(claimValue)));
+                    } else {
+                        claimsCollection.add(createSingleValuedClaim(claimType, claimValue));
+                    }
+                    break;
+                }
+            }
+        }
+        return claimsCollection;
+    }
+
+    /**
+     * Creates a claim with a single value.
+     *
+     * @param claimType  The claim type.
+     * @param claimValue The claim value.
+     * @return The claim.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaim createSingleValuedClaim(String claimType, String claimValue)
+            throws URISyntaxException {
+        ProcessedClaim claim = new ProcessedClaim();
+
+        claim.setClaimType(new URI(claimType));
+        List<Object> claimValues = new ArrayList<>();
+        claimValues.add(claimValue);
+        claim.setValues(claimValues);
+
+        LOGGER.debug("Created claim with type [{}] and value [{}].", claimType, claimValue);
+
+        return claim;
+    }
+
+    /**
+     * Creates a client for sending an AttributeQuery
+     * request to a specified external attribute store.
+     *
+     * @param simpleSign                to create signature for request
+     * @param externalAttributeStoreUrl endpoint of external web service
+     * @param issuer                    of request
+     * @param destination               of request
+     * @return AttributeQueryClient
+     */
+    protected AttributeQueryClient createAttributeQueryClient(SimpleSign simpleSign,
+            String externalAttributeStoreUrl, String issuer, String destination) {
+        return new AttributeQueryClient(simpleSign, externalAttributeStoreUrl, issuer, destination);
+    }
+
+    public void setSimpleSign(SimpleSign simpleSign) {
+        this.simpleSign = simpleSign;
+    }
+
+    public void setExternalAttributeStoreUrl(String externalAttributeStoreUrl) {
+        this.externalAttributeStoreUrl = externalAttributeStoreUrl;
+    }
+
+    public void setIssuer(String issuer) {
+        this.issuer = issuer;
+    }
+
+    public void setDestination(String destination) {
+        this.destination = destination;
+    }
+
+    public void setAttributeMapLocation(String attributeMapLocation) {
+        if (attributeMapLocation != null && !attributeMapLocation.isEmpty() && !attributeMapLocation
+                .equals(this.attributeMapLocation)) {
+            attributeMap = PropertiesLoader
+                    .toMap(PropertiesLoader.loadProperties(attributeMapLocation));
+        }
+        this.attributeMapLocation = attributeMapLocation;
```

Comment:
This gets set regardless of the above tests? That seems wrong, as it might lead to an `attributeMap` with values not corresponding to the the `attributeMapLocation`.